wrangler/src/@utils/stellar-sdk-utils.js

```diff
@@ -1,84 +1,161 @@
-            for (let i = 0; i < txSignatures.length; i++) {
-                const decSig = txSignatures[i];
-                if (!decSig.hint().equals(keypair.signatureHint())) {
-                    continue;
-                }
-                if (keypair.verify(hashedSignatureBase, decSig.signature())) {
-                    signersFound.add(signer);
-                    txSignatures.splice(i, 1);
-                    break;
-                }
-            }
-        }
-        return Array.from(signersFound);
+    catch (err) {
+      throw new Error("Signer is not a valid address: " + err.message);
     }
-    Utils.gatherTxSigners = gatherTxSigners;
-})(Utils || (Utils = {}));
\ No newline at end of file
+    for (let i = 0; i < txSignatures.length; i++) {
+      const decSig = txSignatures[i];
+      if (!decSig.hint().equals(keypair.signatureHint())) {
+        continue;
+      }
+      if (keypair.verify(hashedSignatureBase, decSig.signature())) {
+        signersFound.add(signer);
+        txSignatures.splice(i, 1);
+        break;
+      }
+    }
+  }
+  return Array.from(signersFound);
+}
+
+
+/**
+ * Process a fee payment made to the Turret
+ * 
+ * @param {Object} env The current node env variables
+ * @param {string} xdr The XDR of the fee payment to submit
+ * @param {string | number} min The minimum fee payment amount
+ * @param {string | number} max The maximum fee payment amount
+ * 
+ * @typedef {Object} PaymentResult
+ * @property {string} hash The resultant hash of the TX
+ * @property {string} amount The amount payed
+ *
+ * @returns {PaymentResult} The result of the payment
+ * @throws If the transaction is unable to be submitted for any reason
+ */
+export async function processFeePayment(env, xdr, min, max) {
+  const { HORIZON_URL, STELLAR_NETWORK, TURRET_ADDRESS } = env
+
+  const transaction = new Transaction(xdr, Networks[STELLAR_NETWORK])
+  const transactionHash = transaction.hash().toString('hex')
+
+  if (transaction.operations.length !== 1) {
+    throw { message: `Fee payments cannot have more than one operation` }
+  }
+
+  const op = transaction.operations[0];
+  if (op.type !== 'payment' || op.destination !== TURRET_ADDRESS || !op.asset.isNative()) {
+    throw { message: `Fee payments must be XLM payments made to ${TURRET_ADDRESS}` }
+  }
+
+  if (min) {
+    if (new BigNumber(op.amount).isLessThan(min)) {
+      throw { message: `Fee payment too low. Min = ${min}` }
+    }
+  }
+
+  if (max) {
+    if (new BigNumber(op.amount).isGreaterThan(max)) {
+      throw { message: `Fee payment too large. Max = ${max}` }
+    }
+  }
+
+  let submissionCheckResult = await fetch(`${HORIZON_URL}/transactions/${transactionHash}`);
+  if (submissionCheckResult.ok) {
+    throw { message: `Fee payment with hash ${transactionHash} has already been submitted` }
+  } else if (submissionCheckResult.status === 404) {
+    let txHash = await processTransaction(HORIZON_URL, transaction);
+    return {
+      hash: txHash,
```

Comment:
```suggestion
    await processTransaction(HORIZON_URL, transaction);
    return {
      hash: transactionHash,
```

Transaction hash doesn't change after submission.